test/inbox_SUITE.erl

```diff
@@ -0,0 +1,195 @@
+         outgoing_other_marker_should_not_change_unread_counter/1,
+         has_chat_marker_tests/1
+        ]).
+
+all() ->
+    [
+     {group, one2one},
+     {group, muclight},
+     {group, muc},
+     has_chat_marker_tests
+    ].
+
+groups() ->
+    [
+     {one2one, [], check_marker_activity()},
+     {muclight, [], check_marker_activity()},
+     {muc, [], check_marker_activity()}
+    ].
+
+check_marker_activity() ->
+    [
+     incoming_any_marker_should_not_affect_inbox,
+     outgoing_any_marker_should_not_be_written_to_inbox,
+     outgoing_reset_marker_should_reset_unread_counter,
+     outgoing_other_marker_should_not_change_unread_counter
+    ].
+
+
+%%%===================================================================
+%%% Suite setup/teardown
+%%%===================================================================
+init_per_suite(Config) ->
+    {ok, _} = application:ensure_all_started(stringprep),
+    Config.
+
+end_per_suite(_Config) ->
+    ok.
+
+%%%===================================================================
+%%% Group specific setup/teardown
+%%%===================================================================
+init_per_group(one2one, Config) ->
+    [{module, mod_inbox_one2one}| Config];
+init_per_group(muclight, Config) ->
+    [{module, mod_inbox_muclight}| Config];
+init_per_group(muc, Config) ->
+    [{module, mod_inbox_muc}| Config].
+
+end_per_group(_,_C) ->
+    [].
+
+%%%===================================================================
+%%% Individual Test Cases
+%%%===================================================================
+
+has_chat_marker_tests(_) ->
+    AllMarkers = chat_markers(),
+    Funs = lists:flatten([
+        [
+         fun() -> ?assert(mod_inbox_utils:has_chat_marker(
+                            marker(M, ?NS_CHAT_MARKERS), [M])) end,
+         fun() -> ?assert(mod_inbox_utils:has_chat_marker(
+                            marker(M, ?NS_CHAT_MARKERS), AllMarkers)) end,
+         fun() -> ?assertNot(mod_inbox_utils:has_chat_marker(
+                               marker(<<"non-valid-marker">>, ?NS_CHAT_MARKERS), AllMarkers)) end,
+         fun() -> ?assertNot(mod_inbox_utils:has_chat_marker(
+                               marker(M, <<"unknown-namespace">>), [M])) end,
+         fun() -> ?assertNot(mod_inbox_utils:has_chat_marker(
+                               marker(M, <<"unknown-namespace">>), AllMarkers)) end
+        ] || M <- AllMarkers]),
+    Ret = lists:map(fun(F) -> F() end, Funs),
+    ?assert(lists:all(fun(X) -> X =:= ok end, Ret)),
+    ?assertEqual(length(Funs), length(Ret)).
+
+incoming_any_marker_should_not_affect_inbox(Config) ->
+    Mod = ?config(module, Config),
+    lists:foreach(
+      fun(Mark) ->
+              ct:log("Testing Marker ~p~n", [Mark]),
+              test_mod_inbox_one2one(Mark, fun Mod:handle_incoming_message/4),
+              ct:log("ok", [])
+      end,
+      chat_markers()
+     ).
```

Comment:
Why just one space of indentation from `lists`? I'd do:

```suggestion
    lists:foreach(fun(Mark) ->
                          ct:log("Testing Marker ~p~n", [Mark]),
                          test_mod_inbox_one2one(Mark, fun Mod:handle_incoming_message/4),
                          ct:log("ok", [])
                  end,
                  chat_markers()).
```

Reply:
It was my vim's autoindentation 😄

Reply:
Yeah, it does not perfectly cover all cases 🤷‍♂